### PR TITLE
FEAT: `spacing` ufunc implementation

### PR DESCRIPTION
**Pull request by SwayamInSync**
## Copilot Summary
This pull request adds support for the `spacing` function for the QuadPrecision dtype, aligning its behavior with NumPy conventions and providing comprehensive tests. The changes implement the core logic for spacing in both quad and long double precision, register the new ufunc with NumPy, update documentation, and introduce a thorough test suite to verify correctness for edge cases and typical values.

### QuadPrecision `spacing` function implementation

* Added `quad_spacing` and `ld_spacing` functions to compute the spacing between a value and its next representable value, handling NaN and infinity per NumPy conventions and preserving the sign of the input. (`quaddtype/numpy_quaddtype/src/ops.hpp`) [[1]](diffhunk://#diff-ad4140fa03d374f246bdc4d441d868fa759788f2d27a7a29ca259b16780f0850R1011-R1047) [[2]](diffhunk://#diff-ad4140fa03d374f246bdc4d441d868fa759788f2d27a7a29ca259b16780f0850R1332-R1363)

### NumPy ufunc integration

* Registered the new `spacing` ufunc for both quad and long double types, making it available as `np.spacing` for QuadPrecision arrays. (`quaddtype/numpy_quaddtype/src/umath/unary_ops.cpp`)

### Documentation update

* Updated the release tracker to mark `spacing` as implemented and tested for both quad and long double dtypes. (`quaddtype/release_tracker.md`)

### Test suite for `spacing`

* Added a comprehensive `TestSpacing` class with parameterized tests covering NaN/infinity, sign preservation, zero, one, negative one, magnitude, subnormal range, and array inputs, ensuring correctness and alignment with NumPy's behavior. (`quaddtype/tests/test_quaddtype.py`)

### CODE REVIEW

File: quaddtype/numpy_quaddtype/src/ops.hpp [hunk 1]
```diff
@@ -1008,6 +1008,43 @@ quad_nextafter(const Sleef_quad *x, const Sleef_quad *y)
...
+        return QUAD_NAN;
+    }
+    
+    // Determine direction based on sign of x
+    Sleef_quad direction;
+    if (Sleef_icmpltq1(*x, QUAD_ZERO)) {
+        // Negative: move toward -inf
+        direction = Sleef_negq1(QUAD_POS_INF);
+    } else {
+        // Positive or zero: move toward +inf
```
Comment (juntyr) [inline quaddtype/numpy_quaddtype/src/ops.hpp]: Just checked and the spacing for -0.0 and +0.0 is the same ... sigh

File: quaddtype/tests/test_quaddtype.py [hunk 2]
```diff
@@ -2255,4 +2255,177 @@ def test_direction(self):
...
+        1.0, -1.0,
+        2.0, -2.0,
+        10.0, -10.0,
+        100.0, -100.0,
+        1e10, -1e10,
+        1e-10, -1e-10,
+        0.5, -0.5,
+        0.25, -0.25,
+    ])
+    def test_positive_magnitude(self, x):
```
Comment (juntyr) [inline quaddtype/tests/test_quaddtype.py]: maybe rename to test_spacing_is_non_zero ?

File: quaddtype/tests/test_quaddtype.py [hunk 3]
```diff
@@ -2255,4 +2255,177 @@ def test_direction(self):
...
+        """Test that spacing always has positive magnitude"""
+        q_x = QuadPrecision(x)
+        result = np.spacing(q_x)
+        
+        q_zero = QuadPrecision(0)
+        # The absolute value should be positive
+        abs_result = np.abs(result)
+        assert abs_result > q_zero, f"|spacing({x})| should be positive, got {abs_result}"
+    
+    def test_smallest_subnormal(self):
```
Comment (juntyr) [inline quaddtype/tests/test_quaddtype.py]: Could you also add a test for the smallest normal and 2*the smallest normal?

File: quaddtype/numpy_quaddtype/src/ops.hpp (hunk 1 above)
Comment (SwayamInSync) [reply quaddtype/numpy_quaddtype/src/ops.hpp]: yeah same as curent numpy
```
In [2]: np.spacing(-0.0)
Out[2]: np.float64(5e-324)

In [3]: np.spacing(0.0)
Out[3]: np.float64(5e-324)
```

**Review by juntyr — APPROVED** [review]
LGTM
